dialogue-java-client/src/main/java/com/palantir/dialogue/CompletableToListenableFuture.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue;
+
+import com.google.common.util.concurrent.ListenableFuture;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+class CompletableToListenableFuture<T> implements ListenableFuture<T> {
```

Comment:
looks a bit different from https://github.com/spotify/futures-extra/blob/master/src/main/java/com/spotify/futures/CompletableToListenableFutureWrapper.java

Reply:
So it looks like the main difference is in the exception handling they need to do. We definitely don't need to deal with the `CancellationException` because we delegate cancellation to the underlying future, whereas they have a completely separate ListenableFuture that tries to mirror the original one.

It looks like the `CompletionException` was handled in [here](https://github.com/spotify/futures-extra/commit/be9d8e4bd1c58efb2d8d137b737c28ebced4366b), so I added some tests, but can't seem to find any issues with the current implementation. Lemme know if you see something I'm missing here.

Reply:
let's try it